x-pack/plugins/cases/public/containers/use_delete_file_attachment.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMutation } from '@tanstack/react-query';
+import { casesMutationsKeys } from './constants';
+import type { ServerError } from '../types';
+import { useRefreshCaseViewPage } from '../components/case_view/use_on_refresh_case_view_page';
+import { useCasesToast } from '../common/use_cases_toast';
+import { deleteFileAttachments } from './api';
+import * as i18n from './translations';
+
+interface MutationArgs {
+  caseId: string;
+  fileId: string;
+  successToasterTitle: string;
+}
+
+export const useDeleteFileAttachment = () => {
+  const { showErrorToast, showSuccessToast } = useCasesToast();
+  const refreshAttachmentsTable = useRefreshCaseViewPage();
+
+  return useMutation(
+    ({ caseId, fileId }: MutationArgs) => {
+      const abortCtrlRef = new AbortController();
+      return deleteFileAttachments({ caseId, fileIds: [fileId], signal: abortCtrlRef.signal });
+    },
+    {
+      mutationKey: casesMutationsKeys.deleteFileAttachment,
+      onSuccess: (_, { successToasterTitle }) => {
+        showSuccessToast(successToasterTitle);
+        refreshAttachmentsTable();
+      },
+      onError: (error: ServerError) => {
+        showErrorToast(error, { title: i18n.ERROR_TITLE });
```

Comment:
The `ERROR_TITLE` equals to `Error fetching data.` We should change it to `ERROR_DELETING` (`Error deleting data`) or have a custom one only for files, "Error deleting file (s)" for example.